parsec/core/gui/workspaces_widget.py

```diff
@@ -89,24 +99,23 @@ def delete_workspace(self, workspace_button):
         show_warning(self, _("Not yet implemented."))
 
     def rename_workspace(self, workspace_button):
-        current_file_path = os.path.join("/", workspace_button.name)
         new_name = get_text(
             self, _("New name"), _("Enter workspace new name"), placeholder=_("Workspace name")
         )
         if not new_name:
             return
         try:
-            self.jobs_ctx.run(
-                self.core.fs.workspace_rename, current_file_path, os.path.join("/", new_name)
-            )
+            workspace_id = workspace_button.workspace_fs.workspace_entry.access.id
+            self.jobs_ctx.run(self.core.fs._user_fs.workspace_rename, workspace_id, new_name)
+            workspace_button.workspace_fs = self.core.fs._user_fs.get_workspace(workspace_id)
```

Comment:
Getting a new workspace object shouldn't be necessary in the long run, but maybe this solves the mutation issue you were talking about? In this case, a comment `TODO: remove later` would be nice.

Reply:
Yeah I realized that while updating workspace sharing and it will be removed.